### PR TITLE
added a models_dict to DataParallel

### PR DESCRIPTION
**Pull request by taylanbil**
keys are devices
This helps w/ fairseq, where model itself isn't used directly inside
`train_loop_fn`, but rather their `Trainer` class, which has a loss and a
model attribute, that needs to be unique per device.

With this change,

* I can access the trainer inside `train_loop_fn`
* I can still instantiate `model_parallel` with a function that returns
  the `model`, i.e. a `torch.nn.Module`

```python
  model_parallel = dp.DataParallel(
      lambda: task.build_model(args), device_ids=DEVICES)
  criteria = {
      device: task.build_criterion(args) for device in DEVICES}
  trainers = {
      device: Trainer(args, task, model, criteria[device])
      for device, model in model_parallel.models_dict.items()
  }

	....

  def train_loop_fn(model, loader, device, context):
    trainer = trainers[str(device)]
    for i, samples in loader:
      _log_output = trainer.train_step(samples)  # applies model to input and more
      xm.optimizer_step(trainer.optimizer)
    stats = fairseq_train.get_training_stats(trainer)
    return tracker, stats
```

### CODE REVIEW

File: torch_xla_py/data_parallel.py [hunk 1]
```diff
@@ -200,6 +200,10 @@ def __init__(self, network, device_ids=None, batchdim=0, drop_last=False):
       self._contexts.append(Context(torch.device(device)))
       self._native_run = True
 
+  @property
+  def models_dict(self):
+    return dict(zip(self._device_ids, self._models))
```
Comment (dlibenzi) [inline torch_xla_py/data_parallel.py]: I'd just have a models() function that returns a list of tuples (IOW drop the dict()).
